.husky/pre-commit

```diff
@@ -1,4 +1,80 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
+echo ""
 
+echo "Running 'forge build' now"
+forge build
+
+# If the build fails, abort the commit
+if [ $? -ne 0 ]; then
+  echo "\nForge build failed. Aborting commit.\n"
+  exit 1
+fi
+
+echo ""
+echo ""
+echo "Running 'yarn lint-staged' now:"
 yarn lint-staged
+
+echo ""
+echo ""
+echo "now checking for .env secrets and private keys accidentally being committed to Github"
+echo "(any 64-byte hex string will be identified as potential private key)"
+echo ""
+
+# Regex pattern to identify potential Ethereum private keys (64 hexadecimal characters)
+ETH_PRIVATE_KEY_PATTERN="[a-fA-F0-9]{64}"
+
+# Load secrets from .env file
+if [ -f ".env" ]; then
+  ENV_SECRETS=$(grep -v '^#' .env | grep -v '^\s*$' | cut -d '=' -f 2- | sed 's/ *#.*//')
+else
+  echo ".env file not found"
+  ENV_SECRETS=""
+fi
+
+printAdvise() {
+  ADVISE="If your commit did contain a secret, undo the commit with 'git reset soft HEAD~1', remove the private key and commit again"
```

Comment:
Please adjust command to `git reset --soft HEAD~1`

Reply:
Do show another message if the commit was aborted, as in this case no commit was added, so a reset is not needed.

Reply:
both done